connection/auth.js

```diff
@@ -0,0 +1,81 @@
+export class Authenticator {
+  constructor(http, creds) {
+    this.http = http;
+    this.creds = creds;
+    this.bearer = "";
+    this.expirationEpoch = 0
+  }
+
+  refresh = () => {
+    if (this.creds instanceof AuthUserPasswordCredentials) {
+      return new UserPasswordAuthenticator(this.http, this.creds, this.getOpenidConfig)
+        .refresh()
+        .then(resp => {
+          this.bearer = resp.bearer;
+          this.expirationEpoch = resp.expirationEpoch;
+        });
+    }
+
+    // TODO: Future authentication flows go here
```

Comment:
Could we also add a mode where users can add the accesstoken directly? This could also be achieved through setting an extra header, but I think having an auth-method would be a better UI.

Reply:
done